core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -655,7 +656,16 @@ public void setDefaultTTL(int defaultTTL) {
                         .doOnNext(i -> insertedDataPointEvents.onNext(metric)))
                 .doOnNext(meter::mark);
 
-        Observable<Integer> indexUpdates = dataAccess.updateMetricsIndex(metrics)
+        Observable<Metric<T>> updateIdxMetrics = metrics.filter(metric -> {
```

Comment:
We can still make tests deterministic by putting some hooks in place. I have do things like this in my job scheduler branch in order to try and make things deterministic. You could do something like this:

```java
private Optional<PublishSubject<Void>> indexUpdatesFinished = Optional.empty();
...
// package access for testing
void setIndexUpdatesFinished(PublishSubject<Void> subject) {
    indexUpdatesFinished = Optional.of(subject);
}

// In the method that has the indexUpdates observable
indexUpdates.subscribe(
    aVoid ->  // nothing really to do here
    t ->   // Need to log the error and for tests notify that the update failed
    () -> indexUpdatesFinished.ifPresent(subject -> subject.onNext(void))
);
```
Your tests pass in the subject and subscribe to it in order to get notification of when the updates finish. It is a little more work, but that's part in parcel when dealing with async code :) Doing the index updates out of band was the main thing I wanted to see happen with this ticket, so I'd like us to try and make whatever testing changes we can to support that.